### DISCUSSION TITLE
How can the raw assets be accessed programmatically in MAUI?

### DISCUSSION THREAD

**Discussion by arivoir** [dotnet/maui#3273] [Q&A]
I'm developing a control that is rendered based on the content of a file provided by a customer. 
The file would be provided as a raw asset in the application.
How can a raw asset be accessed from a control programmatically?

**Comment by shv07** [comment]
Any update on this? The only option I was able to find was to read the raw asset using just its filename (with ```Microsoft.Maui.Storage.FileSystem.OpenAppPackageFileAsync(string filename)```). But what if I need the path to the file to pass it to another service?

**Comment by ifitllc** [comment]
I added a json file foo.json into Resources\Raw,   while debugging on Android device:  
```C#
await  FileSystem.Current.OpenAppPackageFileAsync("foo.json")
```
reported file not found error.   I checked the apk file, foo.json is already packaged into the assets folder.

**Reply by MartyIX** [reply]
I'm not too sure why only async methods are exposed and not sync ones (i'm willing to a wait a bit).

**Reply by MartyIX** [reply]
It's likely because UWP https://learn.microsoft.com/en-us/dotnet/api/system.io.windowsruntimestorageextensions?view=dotnet-uwp-10.0 supports only async method for manipulating files (AFAIK).
